csharp/src/Ice/InputStream.cs

```diff
@@ -1091,13 +1128,35 @@ private void Skip(int size)
             _pos += size;
         }
 
+        private void SkipFixedLengthSize()
+        {
+            if (OldEncoding)
+            {
+                Skip(4);
+            }
+            else
+            {
+                // With the 2.0 encoding, there is only one encoding for size, and the writer decides how many bytes
+                // to use.
+                SkipSize();
```

Comment:
I'll just skip the size directly here, and avoid to check OldEncoding twice here and in SkipSize, or we can use a small helper in both that the compiler can inline

```
byte b = _buffer[_pos];
Skip(1 << (b & 0x03));
```

Reply:
Ok, fixed.